plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -49,20 +51,50 @@
     private AmazonEC2 buildClient(Ec2ClientSettings clientSettings) {
         final AWSCredentialsProvider credentials = buildCredentials(logger, clientSettings);
         final ClientConfiguration configuration = buildConfiguration(logger, clientSettings);
-        final AmazonEC2 client = buildClient(credentials, configuration);
+
+        final AwsClientBuilder.EndpointConfiguration endpointConfiguration;
         if (Strings.hasText(clientSettings.endpoint)) {
             logger.debug("using explicit ec2 endpoint [{}]", clientSettings.endpoint);
-            client.setEndpoint(clientSettings.endpoint);
+            endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(clientSettings.endpoint,
+                buildRegion(clientSettings.endpoint));
+        } else {
+            logger.debug("No endpoint defined. Using default endpoint [ec2.us-east-1.amazonaws.com].");
```

Comment:
@henningandersen @DaveCTurner @dadoonet 

Hello All,

 We are looking for multiple region ec2-discovery option.

  Can you please update where are we with the below  comment. Is it already implemented and in-use or its gonna be in future release ?

**Given that this is potentially a breaking change (with master nodes in us-east-1 and data nodes in some other region, the data nodes will no longer be able to discover the master nodes)**
  
    
Thanks

Reply:
@suganselvaraj this is still outstanding, i.e., not implemented into master or any releases yet.

Reply:
Note that:

> We are looking for multiple region ec2-discovery option.

This is not something we really support AFAIK. You should have all nodes within the same region for latency reasons. 

My 2 cents.

Reply:
@suganselvaraj I think it will not work using "ec2-discovery" because under the hood it's searching for nodes inside the same aws region.

You can use a fixed list of IP's, but it can cause latency issues.